src/julia.jl

```diff
@@ -33,4 +33,77 @@ function multireshape(parent::AbstractArray, dimss::Array{Tuple{Vararg{Int}}})
     multireshape(parent, dimss...)
 end
 
+"""
+    pairwise(f::Function, a[, ::Type{Symmetric}])
+
+Apply the function `f` to all pairwise combinations of elements from `a`. PassPro
+`Symmetric` to only calculate `f(x,y)` when `f(x,y) == f(y,x)` and return a
+`Symmetric Matrix`.
+"""
+function pairwise(f::Function, a)
+    n = length(a)
+    firstval = f(first(a), first(a))
+    r = Matrix{typeof(firstval)}(n, n)
+    pairwise!(f, a, r)
```

Comment:
I have pushed a change so it only is calculated once. It is possible there is a more elegant way to do it, but checking `i != 0 || j != 0` was the best I could think of.